File: JavaLibrary/src/main/java/com/flat/simplex/parser/logic/line/call/CallGroup.java
package com.flat.simplex.parser.logic.line.call;

import com.flat.simplex.lexer.Token;
import com.flat.simplex.parser.logic.Block;
import com.flat.simplex.parser.logic.LineParser;
import com.flat.simplex.parser.logic.error.Error;
import com.flat.simplex.parser.logic.line.LineValue;

public class CallGroup extends LineCall {

    private LineValue lineValue;
    
    public CallGroup(Block parent, Token token) {
        super(parent, token, Type.Group);
    }

    @Override
    public void load() {
        Token start = getToken().getChild();
        Token end = getToken().getLastChild();
        if (start == end) {
            if (end == null) {
                getParent().error(getToken(), Error.missingCloser);
            }
    
            getParent().error(getToken(), Error.lineEmptyLine);
        } else {
            lineValue = new LineParser(getParent(), start, end).parse();

            if (end == null) {
                getParent().error(start, Error.missingCloser);
            }
        }
    }

    public LineValue getLineValue() {
        return lineValue;
    }

    @Override
    public void setNext(LineCall next) {
        super.setNext(next);
        if (next.getType() == Type.Value || getNext().getType() == Type.Struct || getNext().getType() == Type.Function) {
            getParent().error(next.getToken(), Error.lineUnexpectedCall);
        }
    }
}

File: SimplexApp/simplex/lang/parser/logic/line/call/CallGroup.js
var LineParser, Error;
const LineCall = require("simplex/lang/parser/logic/line/call/LineCall.js");

class CallGroup extends LineCall {
    constructor(parent, token) {
        super(parent, token, LineCall.Group);

        this.lineValue = null;
    }
    load() {
        let start = this.getToken().getChild();
        let end = this.getToken().getLastChild();
        if (start === end) {
            if (end === null) {
                this.getParent().error(this.getToken(), Error.missingCloser);
            }
            this.getParent().error(this.getToken(), Error.lineEmptyLine);
        } else {
            this.lineValue = new LineParser(this.getParent(), start, end).parse();
            if (end === null) {
                this.getParent().error(start, Error.missingCloser);
            }
        }
    }

    getLineValue() {
        return this.lineValue;
    }

    setNext(next) {
        super.setNext(next);
        if (next.getType() === LineCall.Value || this.getNext().getType() === LineCall.Struct || this.getNext().getType() === LineCall.Function) {
            this.getParent().error(next.getToken(), Error.lineUnexpectedCall);
        }
    }
}

module.exports = CallGroup;
LineParser = require("simplex/lang/parser/logic/LineParser.js");
Error = require("simplex/lang/parser/logic/error/Error.js");